### PR TITLE
Add istrivial

### PR DESCRIPTION
**Pull request by fingolfin**
Testing whether a group is trivial can often be done much more
efficiently than computing it size. E.g. by testing if all
generators are the identity.

For now I still chose to use `isfinite`﻿and `order` for the default
implementation as that is more straightforward. It also works for groups that
are infinitely generated

In the end, there is no optimal default implementation anyway;
to get even close, one needs many more concepts, like tests for...
- is the order already known (then use it)
- are generators known, and are there finitely many, and can `isone`
  be computed for them? then do so?
- is the order unknown, but in principle computable? then do so and use it!
- are the generators unknown, but computable? then do so and use it!

But that's for future generations :-)

### CODE REVIEW

File: src/groups.jl [hunk 1]
```diff
@@ -143,6 +143,15 @@ function Base.isfinite(G::Group)
...
 end
 
+
+@doc Markdown.doc"""
+    istrivial(G::Group)
+Test whether group $G$ is trivial.
+
+The default implementation is based on `isfinite` and `order`.
+"""
+istrivial(G::Group) = isfinite(G) && isone(order(BigInt, G))
```
Comment (kalmarek) [inline src/groups.jl:153]: well then, we can improve a bit:
```julia
function istrivial(G::Group)
    hasgens(G) && return all(isone, gens(G))
    isfinite(G) && return isone(order(G))
    return false
end
```
hmm... there used to be "suggest" button but I can't find it now? anyway, I'm happy with this!

**Review by kalmarek — CHANGES_REQUESTED** [review]
question: do we export it? it also needs to be added to docs as well

File: docs/src/groups.md [hunk 2]
```diff
@@ -75,14 +75,17 @@ implementing the correct `IteratorSize` (i.e. `Base.HasLength()`, or
...
 
     For practical reasons the largest group you could iterate over in your
     lifetime is of order that fits into an `Int`. For example, $2^{63}$
     nanoseconds comes to 290 years.
+
+## Additional methods
+
+```@docs
+Base.isfinite(G::Group)
+Base.istrivial(G::Group)
```
Comment (kalmarek) [inline docs/src/groups.md:90]: There is no such method as `Base.istrivial`

File: docs/src/groups.md [hunk 3]
```diff
@@ -75,14 +75,17 @@ implementing the correct `IteratorSize` (i.e. `Base.HasLength()`, or
 optimized to work only based on the type of the group. In particular when the
 information is derivable from the type, there is no need to extend
 
-```@docs
```
Comment (kalmarek) [inline docs/src/groups.md:78]: the sentence above is now unfinished. It should read
```
[...] there is no need to extend [`isfinite`](@ref).
```

**Review by kalmarek — CHANGES_REQUESTED** [review]

**Comment by kalmarek** [comment]
@fingolfin do you want to have it for `v0.4`, or should I release immediately?

File: docs/src/groups.md (hunk 2 above)
Comment (fingolfin) [reply docs/src/groups.md]: Indeed, silly me, fixed now

File: docs/src/groups.md (hunk 3 above)
Comment (fingolfin) [reply docs/src/groups.md:78]: Agreed, but that's not part of this PR, is it?

**Comment by fingolfin** [comment]
It's not that important to me if it is in 0.4 or not, but if it still fits in, sure, why not

File: docs/src/groups.md (hunk 3 above)
Comment (kalmarek) [reply docs/src/groups.md:78]: you moved the docstring that finished it, didn't you? ;)

**Review by kalmarek — CHANGES_REQUESTED** [review]

File: docs/src/groups.md (hunk 3 above)
Comment (fingolfin) [reply docs/src/groups.md:78]: Oh, I didn't realize that was what it did. Honestly, I think that's a bad idea. I'll change it.

**Pull request merged**